x/tally/keeper/payout.go

```diff
@@ -10,22 +11,44 @@ import (
 	"github.com/sedaprotocol/seda-chain/x/tally/types"
 )
 
-// CalculateCommitterPayouts constructs distribution messages that
-// pay the fixed gas cost for each commiter of a given data request.
-func (k Keeper) CalculateCommitterPayouts(ctx sdk.Context, req types.Request, gasPrice math.Int) (types.DistributionMessages, error) {
-	result := types.DistributionMessages{
-		Messages:   []types.DistributionMessage{},
-		RefundType: types.DistributionTypeTimedOut,
+// CalculateDataProxyPayouts returns payouts for the data proxies and
+// returns the the gas used by the data proxies per executor.
+func (k Keeper) CalculateDataProxyPayouts(ctx sdk.Context, proxyPubKeys []string, gasPrice math.Int) ([]types.Distribution, uint64) {
+	if len(proxyPubKeys) == 0 {
+		return nil, 0
 	}
-	if len(req.Commits) == 0 {
-		return result, nil
+	gasUsed := math.ZeroInt()
+	dists := make([]types.Distribution, len(proxyPubKeys))
```

Comment:
Should we not pass the capacity as not all 'slots' of the slice are guaranteed to be filled? And change the implementation to use append? Or is it ok to have the calling code be responsible for dealing with empty 'slots'?

Reply:
Great catch, fixed.